rust/template/ddlog.h

```diff
@@ -488,7 +496,7 @@ extern void ddlog_free(ddlog_record *rec);
  * (currently only applicable to the string returned by `ddlog_profile()` and
  * `ddlog_dump_record()`).
  */
-extern void ddlog_string_free(char *s);
+extern void ddlog_string_free(const char *s);
```

Comment:
It shouldn't be `const`, as it actually destroys the string.

Reply:
I think it should be const, even though it destroys the string.
Otherwise I cannot deallocate any const pointer ever.
For example, the pointer that comes from `get_table_name`.

Reply:
A pointer that is supposed to be deallocated by the user must not be `const`. In particular, `get_table_name` should not return a `const` pointer. When we fix the API so that the user does not have to free the string, we will add the `const` qualifier to `get_table_name`, which will force users to stop calling `ddlog_string_free`.

Reply:
Actually we do want to pass a const pointer to the user, because we don't want the user to modify that string. So we should cast it to const. This is kind of stupid really, const should not deal with the pointer lifetime, just with the contents of the memory pointed by. There should be a different mechanism for the lifetime, but we will not fix C today.